frontend/public/components/factory/list-page.jsx

```diff
@@ -246,7 +246,19 @@ export const ListPage = props => {
       href = namespaced ? `/k8s/ns/${namespace || 'default'}/${ref}/new` : `/k8s/cluster/${ref}/new`;
     } catch (unused) { /**/ }
   }
-  const createProps = createHandler ? {onClick: createHandler} : {to: href};
+  let createProps;
```

Comment:
Not sure how the [FireMan_](https://github.com/openshift/console/blob/master/frontend/public/components/factory/list-page.jsx#L163) will evaluate the truly the Dropdown component, since the [ListPage component](https://github.com/openshift/console/blob/master/frontend/public/components/factory/list-page.jsx#L249) sets `createProps` either with **onClick** or **to** keys, there is no **items** key set, so there is no way [Dropdown](https://github.com/openshift/console/blob/master/frontend/public/components/factory/list-page.jsx#L165) will be rendered. Or maybe I'm missing something

Reply:
See the ["Create New" button for the `ClusterServiceVersion-v1` detail view](https://teamui.console.team.coreos.systems/k8s/ns/alm-dev/clusterserviceversion-v1s/etcdoperator.v0.9.2) and [its implementation](https://github.com/openshift/console/blob/master/frontend/public/components/cloud-services/clusterserviceversion-resource.tsx#L97).

Reply:
But the [implementation](https://github.com/openshift/console/blob/master/frontend/public/components/cloud-services/clusterserviceversion-resource.tsx#L97) counts with setting the `items` which is missing in the [ListPage component](https://github.com/openshift/console/blob/master/frontend/public/components/factory/list-page.jsx#L249).

Reply:
I see what you mean. Try this:
1. Allow passing `createProps` as a prop to `ListPage`
2. Change [this line](https://github.com/openshift/console/blob/master/frontend/public/components/factory/list-page.jsx#L249) to: 
```javascript
const createProps = createProps || (createHandler ? {onClick: createHandler} : {to: href});
```
3. Change `SecretsPage` to this:
```javascript
const SecretsPage = props => {
  const createItems = {
    // source: 'Create Source Secret',
    // image: 'Create Image Pull Secret',
    // generic: 'Create Key/Value Secret',
    webhook: 'Webhook Secret',
    yaml: 'Secret from YAML',
  };

  const createProps = {
    items: createItems,
    createLink: (type) => `/k8s/ns/${props.obj.metadata.namespace}/secrets/${type === 'yaml' ? 'new' : type}`
  };

  return <ListPage ListComponent={SecretsList} canCreate={true} rowFilters={filters} createButtonText="Create" createProps={createProps} {...props} />;
};
```
I tried this locally and it works.